client/src/chat/ReactionList.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react";
+
+export function ReactionList({
+  reactions,
+  players,
+  currentPlayerPosition,
+  onRemove,
+  align = "right",
+}) {
+  const [hoveredReaction, setHoveredReaction] = useState(null);
+
+  if (!reactions || reactions.length === 0) return null;
+
+  // Group reactions by emoji
+  const groupedReactions = reactions.reduce((acc, reaction) => {
+    const { emoji } = reaction;
+    if (!acc[emoji]) {
+      acc[emoji] = [];
+    }
+    acc[emoji].push(reaction);
+    return acc;
+  }, {});
+
+  const getPlayerName = (playerPosition) => {
+    const player = players?.find((p) => p.get("position") === playerPosition);
+    return player?.get("name") || `Player ${playerPosition + 1}`;
+  };
+
+  const alignmentClass = align === "right" ? "justify-end" : "justify-start";
+
+  return (
+    <div className={`flex gap-1 flex-wrap ${alignmentClass} -mt-2`}>
+      {Object.entries(groupedReactions).map(([emoji, emojiReactions]) => {
+        const count = emojiReactions.length;
+        const hasCurrentPlayer = emojiReactions.some(
+          (r) => r.playerPosition === currentPlayerPosition
+        );
+        const currentPlayerReaction = emojiReactions.find(
+          (r) => r.playerPosition === currentPlayerPosition
+        );
+
+        return (
+          <div
+            key={emoji}
+            className="relative inline-block"
+            onMouseEnter={() => setHoveredReaction(emoji)}
+            onMouseLeave={() => setHoveredReaction(null)}
+          >
+            <button
+              type="button"
+              className={`inline-flex items-center gap-1 px-2 py-0.5 rounded-full text-sm ${
+                hasCurrentPlayer
+                  ? "bg-blue-100 border-2 border-blue-400"
+                  : "bg-gray-100 border border-gray-300"
+              } hover:bg-gray-200 transition-colors`}
+              onClick={() => {
+                if (hasCurrentPlayer && onRemove && currentPlayerReaction) {
```

Comment:
For some reason, the emoji isn't getting removed when you click on it. The button does activate, and successfully passes the remove action up to Chat, and saves to the actions list.